src/transformers/models/deberta/modeling_deberta.py

```diff
@@ -765,16 +860,20 @@ def __init__(self, config):
     def forward(self, input_ids=None, token_type_ids=None, position_ids=None, mask=None, inputs_embeds=None):
         if input_ids is not None:
             input_shape = input_ids.size()
+            device_discriminator = input_ids
+            using_ids = True
         else:
             input_shape = inputs_embeds.size()[:-1]
+            device_discriminator = inputs_embeds
+            using_ids = False
 
         seq_length = input_shape[1]
 
         if position_ids is None:
             position_ids = self.position_ids[:, :seq_length]
 
         if token_type_ids is None:
-            token_type_ids = torch.zeros(input_shape, dtype=torch.long, device=self.position_ids.device)
+            token_type_ids = get_token_type_ids_on_device(device_discriminator, using_ids)
```

Comment:
if we do this after computing the word embedding we can always use the same device discriminator no ?